File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/caraccident/CarAccidentCreationRequest.java
package ru.vitaliy.petrov.server.forms.requests.caraccident;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;
import java.util.List;

@Data
public class CarAccidentCreationRequest {

    @NotNull
    @NotEmpty
    private final String carAccidentScene;

    @NotNull
    private final Date carAccidentDate;

    @NotNull
    private final Time carAccidentTime;

    @NotNull
    @NotEmpty
    private final List<Long> carAccidentSynchronizedParticipants;

    @NotNull
    @NotEmpty
    private final List<Long> carAccidentParticipantsVehicles;
}

File: 3.client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/accident/CarAccidentCreationRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.accident

import java.sql.Date
import java.sql.Time

data class CarAccidentCreationRequest(
    val carAccidentScene: String,
    val carAccidentDate: Date,
    val carAccidentTime: Time,
    val carAccidentSynchronizedParticipants: List<Long>,
    val participantsVehicles: List<Long>
)
